sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -143,23 +114,42 @@ def fit(self, X, y=None, sample_weight=None):
         self : returns an instance of self.
         """
         # Convert data
-        X, y = check_X_y(X, y, accept_sparse=("csr", "csc"), multi_output=True,
-                         y_numeric=True)
         if sample_weight is not None and not isinstance(sample_weight, float):
             sample_weight = check_array(sample_weight, ensure_2d=False)
 
         K = self._get_kernel(X)
         alpha = np.atleast_1d(self.alpha)
 
+        copy = self.kernel == "precomputed"
+
         ravel = False
-        if len(y.shape) == 1:
-            y = y.reshape(-1, 1)
-            ravel = True
+        if self._estimator_type == "regressor":
+            X, y_ = check_X_y(X, y, accept_sparse=("csr", "csc"),
+                              multi_output=True, y_numeric=True)
+            if len(y_.shape) == 1:
+                y_ = y_.reshape(-1, 1)
+                ravel = True
+        elif self._estimator_type == "classifier":  # Multiclass classification
+            X, y = check_X_y(X, y)
+            # Check alpha
+            if self.alpha <= 0:
+                raise ValueError('alpha must be positive')
+
+            # y is going to be encoded by zero arrays
+            # with 1 in a position assigned to a label
+            self.label_encoder_ = LabelBinarizer(neg_label=-1, pos_label=1)
+            y_ = self.label_encoder_.fit_transform(y)
+            self.classes_ = self.label_encoder_.classes_
+            if len(self.classes_) == 2:
+                ravel = True
+        else:
+            raise ValueError('BaseKernelRidge cannot be used as'
```

Comment:
If you use python's abc library, the class won't be instantiable

Reply:
I think I don't understand pretty much abc library, but I've implemented it ( 5b6d838 ). However, `_BaseKernelRidge` was able to be instanced. Because I tested in a ipython console:

```python
>>> from sklearn.kernel_ridge.kernel_ridge import _BaseKernelRidge
>>> a = _BaseKernelRidge()
>>> a
_BaseKernelRidge(alpha=1, coef0=1, degree=3, gamma=None, kernel='linear',
         kernel_params=None)
```
so `_BaseKernelRidge` can be instanced. What I added was that this class have the property `_estimator_type = None`, so now 'if/elif/else'  makes sense. Before, because `_estimator_type` was not defined in `_BaseKernelRidge`, an error was raised.